exasol/analytics/query_handler/graph/stage/sql/execution/query_handler_state.py

```diff
@@ -58,20 +59,35 @@ def __init__(
         self._current_query_handler_context: Optional[ScopeQueryHandlerContext] = None
         self._create_current_query_handler()
 
-    def _check_is_valid(self):
-        if self._current_query_handler is None:
-            raise RuntimeError("No current query handler set.")
-
     def get_current_query_handler(
         self,
     ) -> QueryHandler[List[SQLStageInputOutput], SQLStageInputOutput]:
-        self._check_is_valid()
-        return self._current_query_handler
+        value = self._current_query_handler
+        if value is None:
+            raise RuntimeError("No current query handler set.")
+        return value
+
+    @property
+    def current_query_handler_context(self) -> ScopeQueryHandlerContext:
+        value = self._current_query_handler_context
+        if value is None:
+            raise UninitializedAttributeError(
+                "Current query handler context is undefined."
+            )
+        return value
+
+    @property
+    def current_stage(self) -> SQLStage:
```

Comment:
Property method `current_stage` checks the private attribute `_current_stage` and raises an exception when it is `None` to avoid code duplication.

Questions: 
* Q1) The names of the private attribute and the property differ on in the prefix `_` and potentially could be mixed up. Should we use different names that can easier be distinguished? 
* Q2) The property is public. Should it be private?